darshan-util/pydarshan/darshan/tests/test_lib_accum.py

```diff
@@ -0,0 +1,211 @@
+    # in the control flow for `log_get_derived_metrics()`?
+    ("e3sm_io_heatmap_only.darshan",
+     "APMPI",
+     "KeyError"),
+    ("skew-app.darshan",
+     "POSIX",
+     "I/O performance estimate (at the POSIX layer): transferred 41615.8 MiB at 157.49 MiB/s"),
+    ("skew-app.darshan",
+     "MPI-IO",
+     "I/O performance estimate (at the MPI-IO layer): transferred 41615.8 MiB at 55.22 MiB/s"),
+])
+def test_derived_metrics_bytes_and_bandwidth(log_path, mod_name, expected_str):
+    # test the basic scenario of retrieving
+    # the total data transferred and bandwidth
+    # for all records in a given module; the situation
+    # of accumulating drived metrics with filtering
+    # (i.e., for a single filename) is not tested here
+
+    log_path = get_log_path(log_path)
+    if expected_str == "RuntimeError":
+        with pytest.raises(RuntimeError,
+                           match=f"{mod_name} module does not support derived"):
+            log_get_bytes_bandwidth(log_path=log_path,
+                                    mod_name=mod_name)
+    elif expected_str == "ValueError":
+        with pytest.raises(ValueError,
+                           match=f"{mod_name} is not in the available log"):
+            log_get_bytes_bandwidth(log_path=log_path,
+                                    mod_name=mod_name)
+    elif expected_str == "KeyError":
+        with pytest.raises(KeyError, match=f"{mod_name}"):
+            log_get_bytes_bandwidth(log_path=log_path,
+                                    mod_name=mod_name)
+    else:
+        actual_str = log_get_bytes_bandwidth(log_path=log_path,
+                                             mod_name=mod_name)
+        assert actual_str == expected_str
+
+
+@pytest.mark.parametrize("log_name, module, expected", [
+    # we try to match the "File Count Summary"
+    # tables from the old Perl reports, but
+    # expected values for file counts
+    # are from darshan-parser --file
+    # because of issues like gh-867
+
+    # this also means that the average size
+    # column on the old Perl reports cannot always
+    # be relied upon, since that is calculated
+    # using the file counts; furthermore,
+    # total_max_offset_bytes is not printed by
+    # darshan-parser --file, so the avg size column
+    # is not checked quite as robustly as file count
+    # and max size, though in cases where the Perl
+    # report happens to match the file count, it does
+    # seem to match
+
+    # futhermore, the old Perl report doesn't print out
+    # the file count summary table for all modules, for
+    # example often only showing for POSIX, so in those
+    # cases we really just verify the file count and
+    # the other columns are regression guards against
+    # what we currently have (max size may be available
+    # in a subset of these cases as well)
+    ("e3sm_io_heatmap_only.darshan",
+     "POSIX",
+     # <file count> <avg size> <max size>
+     [[3, "100 GiB", "298 GiB"],
```

Comment:
Not actionable yet, but I want to have a closer look at some of these test cases to make sure the values are all expected and make sense. This case, for instance, actually has less "total files" reported here than in the Perl report, which I'm not sure I understand at all -- other examples of mismatches we've seen in the past were with the Perl report having _less_ files than reported by the parser (since it's doing some filtering), but don't think I expected to see _more_ files reported by Perl summary.

Reply:
I don't think anything is actually wrong here. I forgot that the Perl report is summing these file statistics across POSIX and MPI-IO modules, so it's not surprising that if we only sum stats across the POSIX module here that there will be some mismatches.